sysml.library/Domain Libraries/Quantities and Units/VectorFunctions.sysml

```diff
@@ -0,0 +1,12 @@
+package VectorFunctions {
+	import ScalarValues::*;
+    import Quantities::*;
+
+    /** Vector addition and subtraction */
+	calc def '+' specializes NumericalFunctions::'+' (x: VectorQuantityValue, y: VectorQuantityValue): VectorQuantityValue;
+	calc def '-' specializes NumericalFunctions::'-' (x: VectorQuantityValue, y: VectorQuantityValue): VectorQuantityValue;
+
+    /** Scalar-vector multiplication */
+	calc def '*' specializes NumericalFunctions::'*' (x: Real, y: VectorQuantityValue): VectorQuantityValue;
+	calc def '*' specializes NumericalFunctions::'*' (x: VectorQuantityValue, y: Real): VectorQuantityValue;
```

Comment:
The overloaded calc def of `*` is marked as warning.  @seidewitz could you please help us fix this?

Reply:
> The overloaded calc def of `*` is marked as warning. @seidewitz could you please help us fix this?

It is still a TODO to allow overloaded behaviors in a package. The member distinguishibility validation currently requires that all names in a namspace be distinct. However, even though we have not yet implemented the invocation of function specializations based on argument type, the idea is that this won't depend on name overloading but the explicit specialization hierarchy. Therefore, it won't be necessary for the specialized functions to have the name `'*'` to be usable with the `*` infix notation, only for them to be subtypes of the most general `'*'` function that is invoked by that notation.

I have pushed a commit in which I changed the name of these two calc defs to `scalarVectorMult` and `vectorScalarMult`, and added a `'*'` alias for the first one. (I also renamed `VectorFunctions` to `VectorCalculations`, because this is consistent with SysML-level terminology and with the name of the `QuantityCalaculations` package.)